metapi/metapi.js

```diff
@@ -110,17 +110,34 @@ window.metapi = {};
       optionString += "&" + optionNames[i] + "=" + options[optionNames[i]];
     }
 
-    var cached = metapi.postCache.get(ident);
-    if (cached && !overwrite) {
-      return new metapi.Response(true, cached);
+    if (!(ident instanceof Array)) {
+      ident = [ident];
     }
 
+    // Filter duplicates
+    ident = ident.sort().filter(function (item, pos, ary) {
+      return !pos || item !== ary[pos - 1];
+    });
+
+    var response = [];
+    var toLoad = [];
+
+    for (var j = 0; j < ident.length; j++) {
+      var cached = metapi.postCache.get(ident[j]);
+      if (cached && !overwrite) {
+        response.push(new metapi.Response(true, cached));
+      } else {
+        toLoad.push(ident[j]);
+      }
+    }
+
+    var ids = toLoad.join(encodeURIComponent(";"));
     var fetchUrl = "";
-    if (typeof ident === "string") {
-      fetchUrl = "https://metasmoke.erwaysoftware.com/api/posts/urls?urls=" + ident + "&key=" + key + optionString;
+    if (typeof ids[0] === "string") {
+      fetchUrl = "https://metasmoke.erwaysoftware.com/api/posts/urls?urls=" + ids + "&key=" + key + optionString;
     }
-    else if (typeof ident === "number") {
-      fetchUrl = "https://metasmoke.erwaysoftware.com/api/posts/" + ident + "?key=" + key + optionString;
+    else if (typeof ids[0] === "number") {
+      fetchUrl = "https://metasmoke.erwaysoftware.com/api/posts/" + ids + "?key=" + key + optionString;
```

Comment:
Does `... + ids + ...` actually work, or do you need to semicolon-join it first?